### ISSUE TITLE
PHP 7 RealTimeClient connection issue

### ISSUE THREAD

**Issue by rb-cohen** [coderstephen/slack-client#14]
Hi,

Lib been working really well in PHP5. Gave my app a go on PHP 7 today but couldn't get it to work. As far as I can tell everything is the same, except the PHP version, has anyone else had success running the RealTimeClient in PHP7?

The WebSocket appears to be connecting, but nothing happens after that. Waiting for ->on('hello') or ->on('error') nothing happens. Eventually the react loop seems to stop and the script exits.

Will try to debug further, but not had much luck pinpointing it so far. Throwing this out there in case anyone has any suggestions?

**Comment by coderstephen** [comment]
I can't verify this yet, but it could very well be an incompatibility with [Devristo/phpws](https://github.com/Devristo/phpws) and PHP7. This is the package I have been using for WebSocket functionality, but the package has been sort of unmaintained for a while and has some outstanding issues.

I can look into this more, but I don't plan on fixing this issue if that is the source of the problem, as I plan on rewriting the backend of the package to use Icicle anyway. See #10.

**Comment by rb-cohen** [comment]
Thanks Stephen, I'll try to take another look with fresh eyes tomorrow.
Hopefully it's something trivial.

Saw #10, sounds interesting!

**Comment by rb-cohen** [comment]
Yup, looks like an issue in Devristo/phpws. The connection is fine, and messages are received, but as far as I can tell they aren't being decoded properly. None of the "frames" are ever marked as final, and so the messages aren't emitted to slack-client.

Trying to figure out which bit is going wrong, and will open an issue upstream.

This is what I'm seeing with some hacky debugging messages in Devristo\Phpws\Protocol\WebSocketTransportHybi:
```
string(19) "connector handshake"
string(27) "connector transport connect"
string(10) "handleData"
string(18) "�{"type":"hello"}"
string(27) "creating message from frame"
string(20) "message taking frame"
string(20) "message taking frame"
string(20) "message taking frame"
string(20) "message taking frame"
string(20) "message taking frame"
string(20) "message taking frame"
string(20) "message taking frame"
string(20) "message taking frame"
string(10) "handleData"
string(67) "�A{"type":"presence_change","user":"U0FBK5BJQ","presence":"active"}"
string(20) "message taking frame"
string(20) "message taking frame"
string(20) "message taking frame"
string(20) "message taking frame"
string(20) "message taking frame"
```

**Comment by rb-cohen** [comment]
Found the issue, list() does not unpack in PHP7. Opened issue upstream https://github.com/Devristo/phpws/pull/76

If that project is dead, would you consider using a forked version until #10?

(Many thanks for your help and the lib btw!)

**Comment by coderstephen** [comment]
Yeah, I'd be willing to use a forked version. It's not like the original project even has any stable releases anyway...

**Comment by rb-cohen** [comment]
I'll run the tests and then maybe slack-client could use https://github.com/rb-cohen/phpws for now? I suppose I'd need to rename the composer name for packagist and tag a version?

I'll look at removing the Zend\Logger requirements first too.

**Comment by leoi11** [comment]
This should be fixed by running `composer update Devristo/phpws` to update `composer.lock`. I don't have time to do that right now.
